src/uu/tr/src/operation.rs

```diff
@@ -672,18 +685,28 @@ where
     R: BufRead,
     W: Write,
 {
-    let mut buf = [0; 8192];
-    let mut output_buf = Vec::new();
+    let mut buf = [0; 32768]; // Large buffer for better throughput
+    let mut output_buf = Vec::with_capacity(32768); // Pre-allocated output buffer
```

Comment:
Avoid magic number duplication?

```suggestion
    let mut output_buf = Vec::with_capacity(buf.len()); // Pre-allocated output buffer
```